### DISCUSSION TITLE
Gatsby "client-side" environment variables

### DISCUSSION THREAD

**Discussion by xmflsct** [vercel/vercel#3979] [General] [answered]
For Gatsby build time environment variables, all are fine.

But I would need to have ["client-side"](https://www.gatsbyjs.org/docs/environment-variables/#client-side-javascript) for like reCAPTCHA site key.

For my local development environment, I have `.env` for serverless function variables. And I have `.env.build` [as stated](https://zeit.co/docs/v2/build-step#during-local-development). From `.env.build`, Gatsby build time variables work, but not the "client-side" ones used in pages.

Same goes for actual building and I have proper secrets set. Any ideas?

**Comment by styfle** [comment, marked as answer]
Vercel doesn't know the concept of "client-side" because this is determined by your Static Site Generator, in this case Gatsby.

If you need environment variables on the client side, the you'll have to read from the build env vars and probably write a file to the build output, perhaps `data.json`, or something like that which you can then read from the client side.

https://vercel.com/docs/environment-variables

**Reply by xmflsct** [reply]
I figure out that I need to prefix it with `GATSBY_` for it to be available to client side js. Reference at https://www.gatsbyjs.org/docs/environment-variables/#example-of-using-an-environment-variable. For others who have similar question.
